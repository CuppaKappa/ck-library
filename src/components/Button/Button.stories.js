import React from 'react';
import { Button } from './Button';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'CK/Button',
  component: Button,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    // label: { control: 'text' }
  }
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
// const Template = (args) => (
//   <div className='flex flex-col'>
//     <h3>Default</h3>
//     <Button {...args}>{args.label}</Button>

//     <h3 className='mt-8'>Hover</h3>
//     <Button {...args} isHovered>
//       {args.label}
//     </Button>

//     <h3 className='mt-8'>Active</h3>
//     <Button {...args} isActive>
//       {args.label}
//     </Button>
//   </div>
// );

const Template = (args) => <Button {...args}>{args.label}</Button>;

// More on args: https://storybook.js.org/docs/react/writing-stories/args
export const MediumButton = Template.bind({});
MediumButton.args = {
  label: 'Medium Button'
};

export const AltButton = Template.bind({});
AltButton.args = {
  variant: 'alt',
  label: 'Alt Button'
};

export const LinearBlackButton = Template.bind({});
LinearBlackButton.args = {
  variant: 'linearBlack',
  label: 'Linear Black Button'
};

export const LinearWhiteButton = Template.bind({});
LinearWhiteButton.args = {
  variant: 'linearWhite',
  label: 'Linear White Button'
};

export const LargeButton = Template.bind({});
LargeButton.args = {
  variant: 'large',
  label: 'Large Button'
};

export const JumboButton = Template.bind({});
JumboButton.args = {
  variant: 'jumbo',
  label: 'Jumbo Button'
};
