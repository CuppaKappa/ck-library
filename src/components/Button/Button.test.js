import React from 'react';
import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { Button } from './Button';

afterEach(cleanup);

describe('ButtonComponent', () => {
  it('renders correctly', () => {
    const { getByRole } = render(<Button />);
    expect(getByRole('button')).toBeTruthy();
  });

  it('renders correctly with label', () => {
    const { getByText } = render(<Button label='test' />);
    expect(getByText(/test/i)).toBeTruthy();
  });

  it('renders correctly with disabled and cannot be used', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Button onClick={onClick} disabled />);
    userEvent.click(getByRole('button'));
    expect(onClick).not.toHaveBeenCalled();
  });

  it('renders correctly with type', () => {
    const { getByRole } = render(<Button type='submit' />);
    expect(getByRole('button').getAttribute('type')).toBe('submit');
  });

  it('renders correctly with className', () => {
    const { getByRole } = render(<Button className='test' />);
    expect(getByRole('button').className).toBe('test');
  });

  it('renders correctly with style', () => {
    const { getByRole } = render(<Button style={{ color: 'red' }} />);
    const root = getByRole('button');
    expect(window.getComputedStyle(root).color).toBe('red');
  });

  it('renders correctly with children', () => {
    const { getByText } = render(<Button>test</Button>);
    expect(getByText(/test/i)).toBeTruthy();
  });
});

describe('ButtonComponent variants', () => {
  it('renders variant alt', () => {
    const tree = renderer.create(<Button variant='alt' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders variant linearBlack', () => {
    const tree = renderer.create(<Button variant='linearBlack' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders variant linearWhite', () => {
    const tree = renderer.create(<Button variant='linearWhite' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders variant large', () => {
    const tree = renderer.create(<Button variant='large' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders variant jumbo', () => {
    const tree = renderer.create(<Button variant='jumbo' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
