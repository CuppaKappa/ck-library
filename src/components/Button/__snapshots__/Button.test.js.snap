// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent variants renders variant alt 1`] = `
<button
  className="button alt"
  disabled={false}
  type="button"
/>
`;

exports[`ButtonComponent variants renders variant jumbo 1`] = `
<button
  className="button jumbo"
  disabled={false}
  type="button"
/>
`;

exports[`ButtonComponent variants renders variant large 1`] = `
<button
  className="button large"
  disabled={false}
  type="button"
/>
`;

exports[`ButtonComponent variants renders variant linearBlack 1`] = `
<button
  className="button linearBlack"
  disabled={false}
  type="button"
/>
`;

exports[`ButtonComponent variants renders variant linearWhite 1`] = `
<button
  className="button linearWhite"
  disabled={false}
  type="button"
/>
`;
